src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -2114,6 +2132,13 @@ static int parse_args(vector<const char*>& args, std::ostream *err_msg,
     } else if (ceph_argparse_flag(args, i, "--show-cookie", (char *)NULL)) {
       cfg->show_cookie = true;
     } else if (ceph_argparse_witharg(args, i, &cfg->cookie, "--cookie", (char *)NULL)) {
+    } else if (ceph_argparse_witharg(args, i, &arg_value, "--snap-id", (char *)NULL)) {
+      try {
+	cfg->snapid = boost::lexical_cast<uint64_t>(arg_value);
```

Comment:
No need to involve an additional variable and `lexical_cast` here, let `ceph_argparse_witharg` do the work.  See how e.g. `reattach_timeout` is parsed above.